webapp-client/src/component/CategoryList.js

```diff
@@ -31,42 +34,47 @@ export default class CategoryList extends React.Component {
         categoryList,
       });
     } catch (error) {
-      this.showErrorMessage();
+      this.setState({ hasError: true });
```

Comment:
Why are you opting to set the state manually?

Reply:
ah I don't think this has master pulled.